### ISSUE TITLE
Add exceptions to CC0068 depending on the method attribute

### ISSUE THREAD

**Issue by AdrianoRNascimento** [code-cracker/code-cracker#526]
I'm using xUnit test framework and my test methods are all 'private'.
I understand that a private method that is not referenced is treated as 'not used'. My suggestion is to check for some attributes.

```csharp
[Fact]
private void MyTestMethod() 
{...}
```

And ContractInvariantMethodAttribute from Microsoft Code Contracts:

```cs
[ContractInvariantMethod]
private void ObjectInvariant()
{
    Contract.Invariant(...);
}
```

And DataMemberAttrbute from WCF which maybe odd but perfectly legal to use with hidden members:

```cs
[DataMember]
private int State { get; set; }
```

**Comment by akamud** [comment]
I wonder what would be the expected action when there's a private method decorated with an attribute that is indeed unused, like:

```C#
[SomeLogging]
private void MyUnusedMethod()
{ ... }
```

Seems like there isn't a perfect solution since we can never be 100% sure.

**Comment by giggio** [comment]
It makes sense. We are already providing some exceptions, we certainly could add those. Any others?

**Comment by AdrianoRNascimento** [comment]
I don't remember any other right now.

**Comment by VPashkov** [comment]
Hi!

Perhaps the following attributes may also be supported?

A ContractInvariantMethodAttribute from Microsoft Code Contracts.

```cs
[ContractInvariantMethod]
private void ObjectInvariant()
{
    Contract.Invariant(...);
}
```

And DataContractAttribute and DataMemberAttrbute from WCF which maybe odd but perfectly legal to use with hidden types and members.

```cs
[DataContract]
class OrderDTO
{
    [DataMember]
    private int State { get; set; }
}
```

**Comment by AdrianoRNascimento** [comment]
Hi @giggio and @VPashkov,
I did some tests and identified that private classes are not marked as "unused" by Code Craker.

**Comment by giggio** [comment]
Ok, I have updated the issue's description so that these cases are also ignored. And removed the private class ones.

**Comment by martinzimmermann** [comment]
I guess xUnits `[Theory]` should also be added to the exceptions?

**Comment by giggio** [comment]
Closed on #653.
